src/cnxml/se.tsx

```diff
@@ -401,14 +401,14 @@ function docref(node: DocumentReference, attrs: CommonAttrs, children: JSX.Node)
 
 function figure(node: Figure, attrs: CommonAttrs, children: JSX.Node[]): JSX.Node {
     function mapChild(child: JSX.Node): JSX.Node {
-        if (child == null || typeof child === 'string' || child instanceof globalThis.Node) {
-            return child
-        }
-
         if (Array.isArray(child)) {
             return child.map(mapChild)
         }
 
+        if (JSX.nodeIsNotJSXElement(child)) {
+            return child
+        }
+
         if (child.name.namespace === JSX.CNXML_NAMESPACE
         && child.name.local === 'figure') {
             return {
```

Comment:
This change is unrelated to the PR.

Reply:
previous `if` statement was not including `ProcessingInstruction` so it was throwing an error since `child.name....` wasn't accessible on `child: Element | ProcessingInstruction`

Reply:
I see. In this case please keep those checks in their original order when possible, this will make the diff smaller.

Reply:
In this case, I thought that it would be simpler to create `nodeIsNotJSXElement` than creating an if statement with 4 branches and possible with new branches if we add new types to `Node`, but this required changing the order